src/healthData/HealthDataManager.js

```diff
@@ -0,0 +1,180 @@
+            var url;
+            if (params.get("testEnvironment")) {
+                url = brackets.config.healthDataTestServerURL;
+            } else {
+                url = brackets.config.healthDataServerURL;
+            }
+
+            $.ajax({
+                url: url,
+                type: "POST",
+                data: jsonData,
+                dataType: "text",
+                contentType: "text/plain"
+            })
+                .done(function () {
+                    result.resolve();
+                })
+                .fail(function (jqXHR, status, errorThrown) {
+                    console.error("Error in sending health data. Response : " + jqXHR.responseText + ". Status : " + status + ". Error : " + errorThrown);
+                    result.reject();
+                })
+                .always(function () {
+                    PreferencesManager.setViewState("lastTimeSendHealthData", (new Date()).getTime());
+                });
+        })
+            .fail(function () {
+                result.reject();
+            });
+
+        return result.promise();
+    }
+
+    /*
+     * Check if health data is to be send to the server. If user has enable the tracking, health data will be send for every 24 hours.
+     * If 24 hours or more than that has been passed, then send health data to the server
+     */
+    function _checkHealthDataExport() {
+        var result = new $.Deferred();
+        var isHDTracking = prefs.get("healthDataTracking");
+        window.clearTimeout(timeoutVar);
+
+        if (isHDTracking) {
+
+            var lastTimeSend = PreferencesManager.getViewState("lastTimeSendHealthData"),
+                currentTime = (new Date()).getTime();
+
+            if (!lastTimeSend) {
+                var randomTime = Math.floor(Math.random() * ONE_DAY);
+                lastTimeSend = currentTime + randomTime;
+                PreferencesManager.setViewState("lastTimeSendHealthData", lastTimeSend);
+            }
+
+            if (currentTime >= lastTimeSend + ONE_DAY) {
+                sendHealthDataToServer()
+                    .done(function () {
+                        result.resolve();
+                    })
+                    .fail(function () {
+                        result.reject();
+                    })
+                    .always(function () {
+                        timeoutVar = setTimeout(_checkHealthDataExport, ONE_DAY);
+                    });
+
+            } else {
+                timeoutVar = setTimeout(_checkHealthDataExport, lastTimeSend + ONE_DAY - currentTime);
+            }
+        } else {
+            result.reject();
+        }
+
+        return result.promise();
+    }
+
+    prefs.on("change", "healthDataTracking", function () {
+        _checkHealthDataExport();
+    });
+
+    window.addEventListener("online", function () {
+        _checkHealthDataExport();
+    });
+
+    window.addEventListener("offline", function () {
```

Comment:
Do we know these events work reliably in CEF?  I thought @jasonsanjose tried to use them for something in E4B and they didn't actually fire when they should have...